packages/core/src/utils/resolveConfig.ts

```diff
@@ -1,13 +1,13 @@
 import { PrismConfig, PrismConfigFactory } from '../types';
 
-export async function resolveConfig<Config, Input>(
+export function resolveConfig<Config, Input>(
   input: Input,
   config: PrismConfig<Config, Input>,
   defaultConfig?: PrismConfig<Config, Input>
-): Promise<Config> {
+): Config {
```

Comment:
Why remove the Promise?
As far as I recall the config could be stored in a file. That's why the interface is async.

Reply:
E.g. some config may be coming from the code, some from request, some from file. They get merged in the end but can be async

Reply:
This is currently not happening and even if it would — this is not the right place to do that. This function would actually fetch and merge. We need to separate these two concerns: grab the sources from whenever you want, let the `resolve` function melt them together accordingly.

Moreover, the fact that this is async has been a very big source of pain while debugging the issue.